src/solidity/stateDecoder.js

```diff
@@ -0,0 +1,330 @@
+   *
+   * @param {Object} type - current type (object returned by the getType function)
+   * @param {Map} storageContent - storage
+   * @param {Object} location - location in the storage { offset, slot }
+   * @return {Array} return the decoded array
+   */
+  decodeArray: function (type, storageContent, location) {
+    return this.decodeArrayItems(type, storageContent, 0, location)
+  },
+
+  /**
+   * decode array items
+   *
+   * @param {Object} type - current type (object returned by the getType function)
+   * @param {Map} storageContent - storage
+   * @param {Object} location - location in the storage { offset, slot }
+   * @return {Array} return the decoded array items
+   */
+  decodeArrayItems: function (type, storageContent, depth, location) {
+    var ret = []
+    var size = type.dim[depth]
+    if (size === 'dynamic') {
+      size = storageContent[formatHexKey(location.slot.toString(16))]
+      size = parseInt(size)
+      var pointer = getDynamicPointer(location)
+      location.offset = 0
+      location.slot = new BN(pointer.replace('0x', ''), 16)
+    }
+    if (type.dim.length - 1 > depth) {
+      depth++
+      for (var k = 0; k < size; k++) {
+        ret.push(this.decodeArrayItems(type, storageContent, depth, location))
+        location = moveNextArray(location)
+      }
+    } else {
+      for (var i = 0; i < size; i++) {
+        ret.push(this.decodeType(type, storageContent, location))
+        location = moveNextItemInArray(type, location)
+      }
+    }
+    return ret
+  },
+
+  /**
+   * decode int/uint
+   *
+   * @param {Object} type - current type (object returned by the getType function)
+   * @param {Map} storageContent - storage
+   * @param {Object} location - location in the storage { offset, slot }
+   * @return {String} return the decoded int/uint
+   */
+  decodeInt: function (type, storageContent, location) {
+    var value = getValue(type, storageContent, location)
+    value = extractValue(value, type, location)
+    var bigNumber = new BN(value.replace('0x', ''), 16)
+    if (type.innerType === 'uint') {
+      return bigNumber.toString(10)
+    } else if (type.innerType === 'int') {
+      if (isNegative(bigNumber)) {
+        return '-' + bigNumber.sub(new BN(new Array(value.length + 1).join('f'), 16)).sub(new BN(1)).toString(10) // return utileth.fromSigned(utileth.toUnsigned(bigNumber)).toString(10)
+      } else {
+        return bigNumber.toString(10)
+      }
+    }
+  },
+
+  /**
+   * decode struct
+   *
+   * @param {Object} type - current type (object returned by the getType function)
+   * @param {Map} storageContent - storage
+   * @param {Object} location - location in the storage { offset, slot }
+   * @return {Map} return the decoded struct
+   */
+  decodeStruct: function (type, storageContent, location) {
+    var ret = {}
+    for (var k in type.members) {
+      var member = type.members[k]
+      location = {
+        slot: add(member.location.slot, location.slot),
```

Comment:
`location` inside the for loop is overriding the `location` decodeStruct argument. renaming it to `member_location` inside the for loop fixes this.